### PR TITLE
http.user_id tag in opentracing spans

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it adds `http.user_id` tag to opentracing spans.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [X] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-7416 Add user_id tag to traces](https://app.clickup.com/t/2549495/SALEOR-7416)

File: saleor/graphql/context.py [hunk 1]
```diff
@@ -58,6 +59,12 @@ def set_app_on_context(request):
 def get_user(request: RequestWithUser) -> Optional[UserType]:
     if not hasattr(request, "_cached_user"):
         request._cached_user = cast(UserType, authenticate(request=request))
+        try:
+            user_id = request._cached_user.id
+        except AttributeError:
+            pass
```
Comment (NyanKiyoshi) [inline saleor/graphql/context.py:65]: In which cases would this exception happen?

**Comment by patrys** [comment]
Why? We explicitly do not trace any information that could be used to profile individual users.

**Comment by tomaszszymanski129** [comment]
@patrys There was a case where we needed to perform an audit of who performed one particular action. There were no logs or events saved so we thought it would be helpful in the future to at least have the user id (like we also trace an app id).

File: saleor/graphql/context.py (hunk 1 above)
Comment (NyanKiyoshi) [reply saleor/graphql/context.py:65]: Can't you just do `getattr`? But it does look like there `id` property available for use: https://github.com/django/django/blob/0dd29209091280ccf34e07c9468746c396b7778e/django/contrib/auth/models.py#L417-L418

Comment (tomaszszymanski129) [reply saleor/graphql/context.py:65]: strange, I will double check

**Comment by patrys** [comment]
@tomaszszymanski129 We can't do that as that would track every step of every user, not just staff members. If you want to trace admin actions, use logging (not tracing tags, as those are sampled, and we only save about 1% of them) and only do it in mutations that require admin privileges.

**Comment by tomaszszymanski129** [comment]
@patrys Sounds reasonable. Closing this one. I will prepare subsequent ticket to improve the logging and add the missing event (and others if I find any).

**Pull request closed without merging**